sources/directconnect/direct-connect-gateway-association.go

```diff
@@ -0,0 +1,156 @@
+package directconnect
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+const (
+	directConnectGatewayIDVirtualGatewayIDFormat = "direct_connect_gateway_id/virtual_gateway_id"
+	virtualGatewayIDFormat                       = "virtual_gateway_id"
+)
+
+func directConnectGatewayAssociationOutputMapper(_ context.Context, _ *directconnect.Client, scope string, _ *directconnect.DescribeDirectConnectGatewayAssociationsInput, output *directconnect.DescribeDirectConnectGatewayAssociationsOutput) ([]*sdp.Item, error) {
+	items := make([]*sdp.Item, 0)
+
+	for _, association := range output.DirectConnectGatewayAssociations {
+		attributes, err := sources.ToAttributesCase(association, "tags")
+		if err != nil {
+			return nil, err
+		}
+
+		item := sdp.Item{
+			Type:            "directconnect-direct-connect-gateway-association",
+			UniqueAttribute: "associationId",
+			Attributes:      attributes,
+			Scope:           scope,
+		}
+
+		// stateChangeError =>The error message if the state of an object failed to advance.
+		if association.StateChangeError != nil {
+			item.Health = sdp.Health_HEALTH_ERROR.Enum()
+		} else {
+			item.Health = sdp.Health_HEALTH_OK.Enum()
+		}
+
+		if association.DirectConnectGatewayId != nil {
+			// +overmind:link directconnect-direct-connect-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway",
+					Method: sdp.QueryMethod_GET,
+					Query:  *association.DirectConnectGatewayId,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Deleting a gateway will change the state of the association
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
+				},
+			})
+		}
+
+		if association.VirtualGatewayId != nil {
+			// +overmind:link directconnect-virtual-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-virtual-gateway",
+					Method: sdp.QueryMethod_GET, // TODO: Sort this on the source
+					Query:  *association.VirtualGatewayId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Deleting a gateway will change the state of the association
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
+				},
+			})
+		}
+
+		items = append(items, &item)
+	}
+
+	return items, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type directconnect-direct-connect-gateway-association
+// +overmind:descriptiveType Direct Connect Gateway Association
+// +overmind:get Get a direct connect gateway association by direct connect gateway ID and virtual gateway ID
+// +overmind:list **Not supported**
```

Comment:
We can just skip this if it's not supported, that's what I've been doing in the past. Soon there will be a standard for this I hope but for now skipping will do

```suggestion
```